File: programmers/Level1/소수만들기.java
import java.util.*;

class Solution {
    int answer = 0;
    
    public int solution(int[] nums) {
        comb(0, 0, 0, nums);
        return answer;
    }
    
    void comb(int start, int count, int sum, int[] nums) {
        if (count == 3) {
            if (isPrime(sum)) answer++;
            return;
        }
        
        for (int i = start; i < nums.length; i++) {
            comb(i + 1, count + 1, sum + nums[i], nums);
        }
    }
    
    boolean isPrime(int n) {
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        
        return true;
    }
}

File: programmers/Level1/소수만들기.py
from itertools import combinations

def isPrime(num):
    if num == 0 or num == 1:
        return False
    else:
        for n in range(2, int(num**0.5)+1):
            if num % n == 0:
                return False
        return True

def solution(nums):
    answer = 0
    comb = list(combinations(nums, 3))
    
    for arr in comb:
        if isPrime(sum(arr)):
            answer += 1

    return answer